src/v3/providers/coingecko/coingecko.ts

```diff
@@ -127,6 +132,67 @@ automatedTokenMappingsSyncDoc.onChange(automatedMappings => {
   }
 })
 
+const coingeckoToCrossChainMapping = async (
+  coingeckoAssets: ReturnType<typeof asCoingeckoAssetResponse>,
+  tokenTypes: StringNullMap
+): Promise<CrossChainMapping> => {
+  const invertPlatformMapping: { [key: string]: string } = {}
+  for (const [key, value] of Object.entries(coingeckoPlatformIdMapping)) {
+    if (value === null) continue
+    invertPlatformMapping[value] = key
+  }
+  const platformPriorityDoc = await dbSettings.get('platformPriority')
+  const platformPriority = asCouchDoc(asNumberMap)(platformPriorityDoc).doc
+
+  const out: CrossChainMapping = {}
+
+  for (const asset of coingeckoAssets) {
+    let destAsset: { destChain: string; edgeTokenId: string } | undefined
+
+    const platforms = Object.entries(asset.platforms)
+    const getPriority = (k: string): number =>
```

Comment:
move this func to the top level of the file

Reply:
It relies on the platformPriority I'll move it above the for/loop